src/Common/Commands.Common/AzurePSCmdlet.cs

```diff
@@ -73,8 +73,15 @@ protected string PSVersion
             {
                 if (string.IsNullOrEmpty(_psVersion))
                 {   
-                    System.Management.Automation.PowerShell ps = System.Management.Automation.PowerShell.Create();
-                    _psVersion = ps.Runspace.Version.ToString();
+                    if(this.Host != null)
+                    {
+                        _psVersion = this.Host.Version.ToString();
+                    }
+                    else
+                    {
+                        System.Management.Automation.PowerShell ps = System.Management.Automation.PowerShell.Create();
```

Comment:
I would say we can return a default value in this case.  This should only occur in tests